modules/format/src/updating/merge-update-operations.ts

```diff
@@ -130,6 +133,55 @@ function mergeMulOperand<OP extends "$mul">(
   return ret;
 }
 
+function mergePullOperand<OP extends "$pull">(
+  operand1: RegularUpdateOperand<OP>,
+  operand2: RegularUpdateOperand<OP>
+): RegularUpdateOperand<OP> {
+  const ret: RegularUpdateOperand<OP> = Object.assign({}, operand1);
+  Object.entries(operand2).forEach(([k, v]) => {
+    if (k in ret) {
+      const query1 = ret[k];
+      const query2 = v;
+      if ("$eq" in query1 && "$eq" in query2) {
+        // case: {anyPath: $eq: anyValue} + {anyPath: $eq: anyValue}
+        const value1 = query1["$eq"];
+        const value2 = query2["$eq"];
+        if (value1 == value2) {
```

Comment:
I think `===` would be better for strict comparison.

Reply:
I realized that these values can be `object`. We should use `deepEqual()` for strict comparison. However, we don't want to add external dependency and if the comparison fails, `$in` operator should work correctly. What do you think?

1. import deepEqual library
2. do nothing
3. implement a rough deepEqual function
4. others

Reply:
Thanks for your idea.

https://docs.mongodb.com/manual/reference/operator/update/pull/
values can be object. See example above. In case an array contains object.

Then, let's import `fast-deep-equal`.

`Array.includes()` should be replaced into `Array.some(v2 => deepEqual(v1, v2)`